src/event/log_event.rs

```diff
@@ -3,80 +3,91 @@ use serde::{Serialize, Serializer};
 use std::{
     collections::{btree_map::Entry, BTreeMap, HashMap},
     convert::{TryFrom, TryInto},
+    fmt::{Debug, Display},
     iter::FromIterator,
 };
-use string_cache::DefaultAtom;
 
 #[derive(PartialEq, Debug, Clone, Default)]
 pub struct LogEvent {
     fields: BTreeMap<String, Value>,
 }
 
 impl LogEvent {
-    pub fn get(&self, key: &DefaultAtom) -> Option<&Value> {
-        util::log::get(&self.fields, key)
+    #[instrument(skip(self, key), fields(key = %key.as_ref()))]
+    pub fn get(&self, key: impl AsRef<str>) -> Option<&Value> {
+        util::log::get(&self.fields, key.as_ref())
     }
 
+    #[instrument(skip(self, key), fields(key = %key.as_ref()))]
     pub fn get_flat(&self, key: impl AsRef<str>) -> Option<&Value> {
         self.fields.get(key.as_ref())
     }
 
-    pub fn get_mut(&mut self, key: &DefaultAtom) -> Option<&mut Value> {
-        util::log::get_mut(&mut self.fields, key)
+    #[instrument(skip(self, key), fields(key = %key.as_ref()))]
+    pub fn get_mut(&mut self, key: impl AsRef<str>) -> Option<&mut Value> {
+        util::log::get_mut(&mut self.fields, key.as_ref())
     }
 
+    #[instrument(skip(self, key), fields(key = %key.as_ref()))]
     pub fn contains(&self, key: impl AsRef<str>) -> bool {
         util::log::contains(&self.fields, key.as_ref())
     }
 
-    pub fn insert<K, V>(&mut self, key: K, value: V) -> Option<Value>
-    where
-        K: AsRef<str>,
-        V: Into<Value>,
-    {
+    #[instrument(skip(self, key), fields(key = %key.as_ref()))]
+    pub fn insert(
+        &mut self,
+        key: impl AsRef<str>,
```

Comment:
Future work we should probably upgrade this to an owned string so we can insert that allocation.